### PR TITLE
Update QR code scanner to handle channel and community links

### LINKED ISSUE #18743: General QR code scanner: channel and community links
# Problem

General QR scanner is already done in this PR: https://github.com/status-im/status-mobile/pull/18677
However, it doesn't currently handle channel and community links. 

The function to handle them is `status-im.contexts.shell.qr-reader.view/on-qr-code-scanned`
The designs are here: https://www.figma.com/file/eDfxTa9IoaCMUy5cLTp0ys/Shell-for-Mobile?type=design&node-id=451-18994&mode=design&t=1lpw0E5EgTf1p0ej-0

### LINKED ISSUE #18857: [Communities] QR-code to share community channel is null
# Bug Report
https://github.com/status-im/status-mobile/pull/18807
## Problem 
[comment]: # (A problem description)

#### Expected behavior
QR-code looks as [designed](https://www.figma.com/file/q3kZPl7OpbB7btFK7HG0Mr/Sharing-for-Mobile?type=design&node-id=2345-300688&mode=design&t=ZZumLRHC4nc4XtcI-0) and contains link to community channel

#### Actual behavior
QR-code to share community channel is null and looks not as designed (size, boarders)
Screenshot 2024-02-16 at 11 37 31

### Reproduction

1. Login to app
2. Go to communities
3. Long tap on any channel OR open any chanel and tap on three dots menu
4. Tap Share QR-code

### Additional Information

- Status version: PR
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### LINKED ISSUE #18935: [Scanner] Nothing happens when users scans community channel QR-code
# Bug Report

## Problem 
[comment]: # (A problem description)
User tries to scan QR-code to join a community channel and gets no result

#### Expected behavior
Community with a shared channel opens

#### Actual behavior

Nothing happens when users scans community channel QR-code

https://github.com/status-im/status-mobile/assets/147824399/a8f4de63-5b30-46d0-bd08-1b34f2dd4f2e

### Reproduction

1. Login as User A
2. Generate channel QR-code as user A
3. Login as user B (on another device)
4. Open QR-code scanner inside the Status app.
5. As User B can QR-code of User A

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by alwx**
Fixes https://github.com/status-im/status-mobile/issues/18743
Fixes https://github.com/status-im/status-mobile/issues/18935
Fixes https://github.com/status-im/status-mobile/issues/18857 (see comment there)

Updates the QR scanner to handle community and channel links by reusing the code for universal links. I personally think that a more generic solution needs to be used for both universal links and QR code scanner but that won't happen until we refactor universal links (the code there is currently partially outdated) & make a plan and think of how exactly this is going to be done.

#### Platforms

- Android
- iOS

##### Functional

- QR code scanner

status: ready

### CODE REVIEW

**Review by Parveshdhull — APPROVED** [review]

**Review by siddarthkay — APPROVED** [review]

**Review by jo-mut — APPROVED** [review]

**Review by clauxx — APPROVED** [review]

**Review by vkjr — APPROVED** [review]

**Review by yqrashawn — APPROVED** [review]

**Review by BalogunofAfrica — APPROVED** [review]

**Review by jo-mut — APPROVED** [review]

**Comment by mariia-skrypnyk** [comment]
Hi @alwx !
Thanks fro your PR.

**ISSUE 1: QR code coloured background moved to the left**

**Additional Info:** iOS/Android
**Devices:** iOS 13 (17.3.1) Samsung A34 (14.0)

**Now**
Screenshot 2024-04-10 at 18 21 13

**Design**
Screenshot 2024-04-10 at 18 22 12

https://www.figma.com/file/q3kZPl7OpbB7btFK7HG0Mr/Sharing-for-Mobile?type=design&node-id=2345-300688&mode=design&t=XPZhjqfoxVygizLY-0

**Comment by alwx** [comment]
@mariia-skrypnyk it's not the issue of this particular PR — it has nothing to do with the design

**Comment by mariia-skrypnyk** [comment]
> @mariia-skrypnyk it's not the issue of this particular PR — it has nothing to do with the design

Ok, I'll report it separately.

**Comment by mariia-skrypnyk** [comment]
Hi @alwx!

QR code for community channel contains link to Community screen and not to Community channel.
Is this a correct behaviour in you PR? 

No implementation of the whole logic designed [here ](https://www.figma.com/file/eDfxTa9IoaCMUy5cLTp0ys/Shell-for-Mobile?type=design&node-id=451-18994&mode=design&t=Scf8lIFQSdjbVj0B-0) 
Is it TBI for now?

Also, we do have such PR Open https://github.com/status-im/status-mobile/pull/19566 can it be useful for you in this case when will be done by Parvesh?

File: src/status_im/contexts/shell/qr_reader/view.cljs [hunk 1]
```diff
@@ -71,13 +71,10 @@
...
-    ;; TODO: https://github.com/status-im/status-mobile/issues/18743
-    nil
-
-    (text-for-url-path? scanned-text router/channel-path)
-    ;; TODO: https://github.com/status-im/status-mobile/issues/18743
-    nil
+    (or
+     (text-for-url-path? scanned-text router/community-with-data-path)
+     (text-for-url-path? scanned-text router/channel-path))
+    (debounce/debounce-and-dispatch [:universal-links/handle-url scanned-text] 300)
```
Comment (ibrkhalil) [inline src/status_im/contexts/shell/qr_reader/view.cljs:77]: nit: Should the 300ms value be a constant to avoid magic numbers?

**Comment by mariia-skrypnyk** [comment]
Hi @alwx !

Let me know if you have answers for me. 🙏

**Pull request merged**